### PR TITLE
(PUP-7141) stop dup of frozen string in 3x converter for master only 

### PR DESCRIPTION
**Pull request by hlindberg**

### CODE REVIEW

**Comment by hlindberg** [comment]
jenkins please test this

File: lib/puppet/provider/package/windows/package.rb [hunk 1]
```diff
@@ -71,7 +71,7 @@ def self.munge(value)
 
     def self.replace_forward_slashes(value)
       if value.include?('/')
-        value.gsub!('/', "\\")
+        value = value.gsub('/', "\\")
```
Comment (MikaelSmith) [inline lib/puppet/provider/package/windows/package.rb]: Did you intend to revert to a copy?

File: lib/puppet/type/user.rb [hunk 2]
```diff
@@ -733,7 +733,8 @@ def generate
           if entry =~ /^~|^%h/ and not home
             raise ArgumentError, _("purge_ssh_keys value '%{value}' meta character ~ or %{home_placeholder} only allowed for users with a defined home directory") % { value: value, home_placeholder: '%h' }
           end
-          entry.gsub!(/^~\//, "#{home}/")
+          # make sure frozen value is duplicated by using a gsub, second mutating gsub! is then ok
+          entry = entry.gsub(/^~\//, "#{home}/")
```
Comment (MikaelSmith) [inline lib/puppet/type/user.rb]: Same here.

File: lib/puppet/provider/package/windows/package.rb (hunk 1 above)
Comment (hlindberg) [reply lib/puppet/provider/package/windows/package.rb]: yes absolutely - it should not mutate

File: lib/puppet/type/user.rb (hunk 2 above)
Comment (hlindberg) [reply lib/puppet/type/user.rb]: and same here

**Comment by MikaelSmith** [comment]
This diff is kind of weird. When I pull it locally and diff against master the only change is in convert_String.

**Pull request merged**